test/unit/org/apache/cassandra/tools/ToolRunner.java

```diff
@@ -49,6 +49,8 @@
 public class ToolRunner implements AutoCloseable
 {
     protected static final Logger logger = LoggerFactory.getLogger(ToolRunner.class);
+
+    public List<String> stdErrRegExpCleaners = Arrays.asList("(?im)^picked up.*\\R");
```

Comment:
should be final and immutable if public, if private should just be final.

Reply:
Icwym you're probably thinking along the lines of `defaultCleanersList` whereas I was thinking along the lines of `cleanersInThisInstance` and that to be modifiable to your liking. That's why I made it public, not static and not final :shrug:. Makes sense or you'd prefer me to change it?

Reply:
I see where you are coming from, but mutable state can still be brittle for tests, so simpler to avoid it in the API.

Since this is public and mutable, anyone is able to modify it
1) public means any other class can have access, and do with it what Is allowed by the object
2) the object is mutable, given the fact that it is public, any class could mutate unexpectedly 
3) you can't add to this list, so "modifiable to your linking" wouldn't work

```
jshell> var v =Arrays.asList(1, 2)
v ==> [1, 2]

jshell> v.add(3)
|  Exception java.lang.UnsupportedOperationException
|        at AbstractList.add (AbstractList.java:153)
|        at AbstractList.add (AbstractList.java:111)
|        at (#3:1)

jshell> v.remove(1)
|  Exception java.lang.UnsupportedOperationException
|        at AbstractList.remove (AbstractList.java:167)
|        at (#4:1)

jshell> v.set(0, 42)
$5 ==> 1

jshell> v
v ==> [42, 2]
```

Given this, it makes sense to be the `defaultCleanersList`, also makes sense to expose so the caller can add to their own list; but if it is exposed, it should be immutable

Reply:
Ok, immutable and public it is :-)